lib/icinga/checkable-notification.cpp

```diff
@@ -173,6 +174,13 @@ static void FireSuppressedNotifications(Checkable* checkable)
 				bool still_applies = checkable->NotificationReasonApplies(type);
 
 				if (still_applies) {
+					switch (type) {
+						case NotificationProblem:
+						case NotificationRecovery:
+							if (stateType == StateTypeSoft)
+								continue;
+					}
```

Comment:
Writing this as a switch statement just adds "enumeration value ‘NotificationDowntimeStart’ not handled in switch [-Wswitch]" compiler warnings. I don't think that this is any better to read than a simple if statement.

Reply:
IMAO the switch is **the** construct for x is a or b. I'm intelligent, the compiler isn’t (and proves that with this nonsense warning\*).

\* Or would **you** add all possible cases in a `switch (fork()) {`?

Reply:
I mean: sure, I could also add a default (which would cover all of the others) with nothing, but that's switch's default.

Reply:
I opt for not introducing new compiler warnings. I mean the code base is currently a mess in that regard, but it would be nice if it wasn't.

> and proves that with this nonsense warning

Maybe you think that way, I think that in summary, compiler warning do more good than harm, even if they aren't perfect in every situation.

> Or would **you** add all possible cases in a `switch (fork()) {`?

If you count `default:` as "all possible cases" then yes, but in general, I wouldn't write it that way as you lose the PID of the child.

If you let me choose between
```c
pid_t pid = fork();
switch (pid) {
case -1:
    // ...
    break;
case 0:
    // ...
    break;
default:
    // ...
}
```
and
```c
pid_t pid = fork();
if (pid == -1) {
    // ...
} else if (pid == 0) {
    // ...
} else {
    // ...
}
```
I'd go for the second one.